tests/check_new_syntax.py

```diff
@@ -63,30 +83,49 @@ def visit_Subscript(self, node: ast.Subscript) -> None:
     class AnnotationFinder(ast.NodeVisitor):
         def __init__(self) -> None:
             self.set_from_collections_abc = False
+            self.context_manager_from_typing = False
 
         def visit_ImportFrom(self, node: ast.ImportFrom) -> None:
             if node.module == "collections.abc":
                 imported_classes = node.names
                 if any(cls.name == "Set" for cls in imported_classes):
                     self.set_from_collections_abc = True
+            if node.module == "typing":
+                imported_classes = node.names
+                if any(cls.name == "ContextManager" for cls in imported_classes):
+                    self.context_manager_from_typing = True
 
             self.generic_visit(node)
 
         def visit_AnnAssign(self, node: ast.AnnAssign) -> None:
-            OldSyntaxFinder(set_from_collections_abc=self.set_from_collections_abc).visit(node.annotation)
+            OldSyntaxFinder(
+                set_from_collections_abc=self.set_from_collections_abc,
+                context_manager_from_typing=self.context_manager_from_typing,
+            ).visit(node.annotation)
```

Comment:
Nit: These are getting a bit repetitive. It might be good to put the repetitive parts into a new method that e.g. instantiates `OldSyntaxFinder`, or instantiates and then calls `.visit()`.

Reply:
Yeah, makes sense